onnx/backend/test/case/node/__init__.py

```diff
@@ -202,3 +202,15 @@ def collect_testcases():  # type: () -> List[TestCase]
     '''
     import_recursive(sys.modules[__name__])
     return _NodeTestCases
+
+
+def collect_specific_testcases(operator_name):  # type: (Text) -> List[TestCase]
+    '''Collect node test cases which include specific operator
+    '''
+    import_recursive(sys.modules[__name__])
+    # only keep those tests related to this operator
+    specific_testcases = []
+    for testcase in _NodeTestCases:
+        if testcase.model.graph.node[0].op_type == operator_name:
+            specific_testcases.append(testcase)
```

Comment:
instead of creating all the test cases and then picking up the ones for a specific operator can you instead create the test cases only for the requested operator... basically in expect method if the node.op_type does not match the request op_type don't create the test data

Reply:
Are you saying extending the arguments for expect to filter op_type? If so, do we need to change a lot of expect functions to add an extra argument?

Actually these testcases are created by `import_module`. So another approach might be changing `import_recursive`, but the test python script name might not include the op_type name. Please check my comments below. Thanks.

Reply:
Maybe we can introduce a global variable in expect function to filter. I will try it.

Reply:
why cant we filter in expect?
The problem with filter while importing the module is we have to make an assumption that python file name will contain op_type... this is very fragile. 

There is no need to change the signature of expect mtd. expect takes in node as the first argument you can get op_type from node as node.op_type and if it does not match the provided op_type then you can filter it.

Reply:
I think without changing the function `expect`'s interface, we cannot get the given op_type in the `expect` function. op_type needs to be given while calling `expect`.

Reply:
Solved. Thanks for the proposal!